contracts/external/cw-payroll-factory/src/contract.rs

```diff
@@ -0,0 +1,100 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Reply, Response, StdResult, SubMsg, WasmMsg,
+};
+
+use cw2::set_contract_version;
+use cw_payroll::msg::InstantiateMsg as PayrollInstantiateMsg;
+use cw_utils::parse_reply_instantiate_data;
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg};
+
+pub(crate) const CONTRACT_NAME: &str = "crates.io:cw-admin-factory";
+pub(crate) const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+pub const INSTANTIATE_CONTRACT_REPLY_ID: u64 = 0;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    _msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("creator", info.sender))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    _deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::InstantiatePayrollContract {
+            instantiate_msg: msg,
+            code_id,
+            label,
+        } => instantiate_contract(env, info, msg, code_id, label),
+    }
+}
+
+pub fn instantiate_contract(
+    _env: Env,
+    info: MessageInfo,
+    instantiate_msg: PayrollInstantiateMsg,
+    code_id: u64,
+    label: String,
+) -> Result<Response, ContractError> {
+    // Instantiate the specified contract with owner as the admin.
+    let instantiate = WasmMsg::Instantiate {
```

Comment:
are there other planned functions of this factory contract other than allowing payroll contracts to be instantiated as their own admins? (I think I remember you saying something about tracking which payroll contracts belong to which DAOs). If that is the case, what is the benefit of using a factory contract vs tracking this information in dao-core?

Reply:
We could track the information in DAO core with items....

Have some commits I'm going to push soon that will clarify the direction.

Reply:
I think this factory pattern makes sense -- a payroll contract does not always have to be used by a DAO so I agree they should be kept separate

Reply:
Makes things simple and easy to reason about, which is a plus IMO.